app/queries/drug_stocks_query.rb

```diff
@@ -218,10 +222,28 @@ def drug_consumption_cache_key
     [
       "#{self.class.name}#drug_consumption",
       @facilities.map(&:id).sort,
+      @include_block_report,
       @for_end_of_month,
       @protocol.id,
       @state,
       CACHE_VERSION
     ].join("/")
   end
+
+  private
+
+  def set_facility_group
+    facility_group_ids = @facilities.pluck(:facility_group_id).uniq
+    throw "All facilities should belong to the same facility group." if facility_group_ids.count > 1
```

Comment:
We should use `raise`. `throw` actually does some subtly different things in Ruby that we usually don't want.